### PR TITLE
[DRAFT] Import dataset arrow

### PR DESCRIPTION
**Pull request by pachadotdev**
### Intent

Adds an option to Import Dataset to read Parquet/Feather files with Arrow. When this is complete, it shall allow easy import with S3/FTP options.

### Approach

I cloned the XLS import menu and started adapting from there.

### Automated Tests

No tests yet.

### QA Notes

The "Import Arrow..." option does nothing at the moment. I would love some assessment to get it running as a 'duplicate' of "Import Excel..." and then I shall edit until it's completed.

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

**Comment by pachamaltese** [comment]
@kevinushey hi, what do you think of this draft? i plan to complete it soon

**Comment by kevinushey** [comment]
@pachamaltese Thank you for putting this together! I'll try to review soon.

**Comment by kevinushey** [comment]
@pachadotdev, I'm very sorry -- this PR dropped off of our radar. Are you willing to resolve the associated conflicts that have popped up and prepare this PR for re-review?

**Comment by pachadotdev** [comment]
> @pachadotdev, I'm very sorry -- this PR dropped off of our radar. Are you willing to resolve the associated conflicts that have popped up and prepare this PR for re-review?

hi @kevinushey 

no worries, apologies here for the delay
I've fixed some indentation unintended changes, but now I can't get the draft menu, at least I got this
Screenshot from 2022-02-18 02-07-39

I adapted from SAV/DTA import to try to do the same for parquet/feather

File: src/cpp/session/modules/SessionDataImportV2.R [hunk 1]
```diff
@@ -516,6 +516,41 @@
...
+         optionTypes[["range"]] <- "character"
+
+         if (!is.null(options[["range"]]) && nchar(options[["range"]]) > 0) {
+            options[["skip"]] <- NULL
+            options[["n_max"]] <- NULL
+         }
+
+         return(list(
+            name = "read_dataset",
+            reference = arrow::read_dataset,
```
[the remark this answers is not in the document]
Comment (romainfrancois) [reply src/cpp/session/modules/SessionDataImportV2.R:547]: Is this `arrow::open_dataset()` instead ?

**Review by romainfrancois — COMMENTED** [review]
Should there be a `From Text (arrow)` using `read_csv_arrow()` and relevant options for it ?

Also perhaps importing arrow format should not be the same menu item as importing parquet. Maybe this could instead be: 

```
--
From Parquet (arrow)
From Arrow (arrow)
```

This way perhaps it would be easier to have the appropriate options for each format: 

````
> args(arrow::read_feather)
function (file, col_select = NULL, as_data_frame = TRUE, ...) 
NULL
> args(arrow::read_parquet)
function (file, col_select = NULL, as_data_frame = TRUE, props = ParquetArrowReaderProperties$create(), 
    ...) 
NULL
````

**Comment by ntluong95** [comment]
Is there a plan to bring this feature to Rstudio @romainfrancois

**Comment by romainfrancois** [comment]
Given the age of this pr, perhaps things have changed in `arrow` since. Maybe it would be useful to synchronize with the voltron team about this. cc @nealrichardson @thisisnic

**Comment by romainfrancois** [comment]
I think we can reasonnably close the pr now, and come back to it for reference if we want the feature at a later time.

**Comment by thisisnic** [comment]
Sorry for the lack of reply - mentioned this internally and then forgot to comment here.  It sounds like it's something which would be great to have, but this PR isn't there, and I don't believe that any of us at Voltron Data have it on our radar as something we plan to submit a PR for any time soon, so it makes sense to close it.

**Pull request closed without merging**